drivers/spi/spi_ll_stm32.c

```diff
@@ -326,6 +326,39 @@ static void spi_stm32_read_next_frame(SPI_TypeDef *spi,
 	}
 }
 
+#if SPI_STM32_FIFO_ENABLED
+static void spi_stm32_send_next_packet(SPI_TypeDef *spi,
+		struct spi_stm32_data *data)
+{
+	uint8_t packet_size = SPI_PACKET_SIZE_GET(data->ctx.config->operation);
+
+	for (int i = 0; i < packet_size; i++) {
+		spi_stm32_send_next_frame(spi, data);
```

Comment:
I don't know how stm32's SPI controllers work, generally fifo interrupt threshold is meant to make sure that the controller is not going to starve of frames to send (if there are way more to be sent, that is). Which means that between the time you get the interrupt and handle it, the controller may have had time to continue sending frames left in the fifo, which in return, means that it went over the threshold so you'll need to feed more frames than the configured threshold. It can also be true in receiving mode in a RX only transaction type where the controller can send the dummy bytes by itself (I don't kmow if that applies to this controller though).

Reply:
> I don't know how stm32's SPI controllers work, generally fifo interrupt threshold is meant to make sure that the controller is not going to starve of frames to send (if there are way more to be sent, that is).

That's my understanding as well, yes.

> Which means that between the time you get the interrupt and handle it, the controller may have had time to continue sending frames left in the fifo, which in return, means that it went over the threshold so you'll need to feed more frames than the configured threshold. 

I understand you are suggesting that, instead of writing frames to the TxFIFO until the packet size is reached, this should keep writing (or reading) frames as long as TXP (which indicates that the FIFO has room for at least one packet more) is set. If that's the case, yes, it could be done that way.

Reply:
yep, make it a while loop. The little extra time spent doing so, is less of an issue for the system than handling more interrupts due to a threshold that could hit more often.

Reply:
Using a while loop here caused several problems produced by the fact that SPI supports transmissions in which tx is bigger than rx or vice-versa. The current implementation takes care of this, defaulting to the classic way of transmitting in case of rx > tx (which causes more problems that require further modifications).

Reply:
I reviewed this and found a clean way to support the rx > tx case, so this is no longer a problem.